libkbfs/folder_branch_ops.go

```diff
@@ -1911,7 +1987,7 @@ func (fbo *folderBranchOps) finalizeMDWriteLocked(ctx context.Context,
 		} else {
 			bid = fbo.bid
 		}
-		err := mdops.PutUnmerged(ctx, md, bid)
+		mdID, err = fbo.doPutWithCancellationCheck(ctx, lState, md, bid)
```

Comment:
There are a few calls after this that use `ctx`, even though it might be canceled now.  I don't think any of them check for a canceled context, but if they did, or if we added something that did, the whole operation would fail anyway.  Maybe `doPutWithCancellationCheck` should return the new context and we can use that for the rest of it.  Technically probably `finalizeMDWriteLocked` should also return the new context for its callers to use, etc.